sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java

```diff
@@ -0,0 +1,268 @@
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.monitor.ingestion.implementation.IngestionUsingDataCollectionRulesAsyncClient;
+import com.azure.monitor.ingestion.implementation.UploadLogsResponseHolder;
+import com.azure.monitor.ingestion.models.UploadLogsError;
+import com.azure.monitor.ingestion.models.UploadLogsOptions;
+import com.azure.monitor.ingestion.models.UploadLogsResult;
+import com.azure.monitor.ingestion.models.UploadLogsStatus;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * The asynchronous client for uploading logs to Azure Monitor.
+ */
+@ServiceClient(isAsync = true, builder = LogsIngestionClientBuilder.class)
+public final class LogsIngestionAsyncClient {
+    private static final ClientLogger LOGGER = new ClientLogger(LogsIngestionAsyncClient.class);
+    private static final String CONTENT_ENCODING = "Content-Encoding";
+    private static final long MAX_REQUEST_PAYLOAD_SIZE = 1024 * 1024; // 1 MB
+    private static final String GZIP = "gzip";
+    private static final JsonSerializer DEFAULT_SERIALIZER = JsonSerializerProviders.createInstance(true);
+
+    private final IngestionUsingDataCollectionRulesAsyncClient service;
+
+    LogsIngestionAsyncClient(IngestionUsingDataCollectionRulesAsyncClient service) {
+        this.service = service;
+    }
+
+    /**
+     * Uploads logs to Azure Monitor with specified data collection rule id and stream name. The input logs may be
+     * too large to be sent as a single request to the Azure Monitor service. In such cases, this method will split
+     * the input logs into multiple smaller requests before sending to the service.
+     * @param dataCollectionRuleId the data collection rule id that is configured to collect and transform the logs.
+     * @param streamName the stream name configured in data collection rule that matches defines the structure of the
+     * logs sent in this request.
+     * @param logs the collection of logs to be uploaded.
+     * @return the result of the logs upload request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UploadLogsResult> upload(String dataCollectionRuleId, String streamName, List<Object> logs) {
+        return upload(dataCollectionRuleId, streamName, logs, new UploadLogsOptions());
```

Comment:
So, this is a pattern I've seen in Storage and I'd like to begin changing it. Instead of calling into `options` based overload with a newed up options bag object could we either have a constant or call into the package-private overload that takes a splayed out options bag? It ends up being more verbose during development but to me it makes reading the flow of calls easier as there isn't as many jumps